src/porepy/numerics/mixed_dim/coupler.py

```diff
@@ -101,13 +112,24 @@ def split(self, gb, key, values):
 
         """
         dofs = np.empty(gb.size(), dtype=int)
-        for _, d in gb:
-            dofs[d['node_number']] = d['dof']
+        for _, data in gb:
+            k_data = self.__get_data(data, data_key)
+            dofs[data['node_number']] = k_data['dof']
         dofs = np.r_[0, np.cumsum(dofs)]
 
-        gb.add_node_prop(key)
+        if data_key is None:
+            gb.add_node_prop(key)
+
         for g, d in gb:
-            i = d['node_number']
-            d[key] = values[slice(dofs[i], dofs[i+1])]
+            k_data = self.__get_data(data, data_key)
+            i = data['node_number']
+            k_data[key] = values[slice(dofs[i], dofs[i+1])]
 
 #------------------------------------------------------------------------------#
+    def __get_data(self, data, data_key):
```

Comment:
The idea is to either return the sub-dictionary, or the dictionary itself if data_key is not found? This should mean that for methods that do not need the subdict, we can just store the parameters in `data`, forget about keywords like `flow`, and everything will work as before?

If correct, this is elegant, but it can lead to errors that are hard to debug. For instance, if I misspell `flow`, or any of a number of other scenarios. I can't see better ways of doing it, but I suggest we take some precautions where possible, such as surrounding L131 with a try-except.